openassessment/xblock/openassessmentblock.py

```diff
@@ -352,6 +352,34 @@ def get_anonymous_user_id(self, username, course_id):
         """
         return self.runtime.service(self, 'user').get_anonymous_user_id(username, course_id)
 
+    def is_user_state_service_available(self):
+        """
+        Check if the user state service is present in runtime.
+        """
+        return self.runtime._services.get('user_state') is not None  # pylint: disable=protected-access
+
+    def get_user_state(self, username):
+        """
+        Get the student module state for the given username for current ORA block.
+
+        Arguments:
+            username(str): username against which the state is required in the current block.
+
+        Returns:
+            user state, if found, else empty dict
+        """
+        user_state_service = self.runtime._services.get('user_state')  # pylint: disable=protected-access
```

Comment:
Same idea here with `@XBlock.needs('user_state')`.

Reply:
Also, it's worth mentioning that this platform PR: https://github.com/edx/edx-platform/pull/22495 is required to test locally.

Reply:
@iloveagent57 Already mentioned in the ticket's description.

Reply:
D'oh, sorry I missed that.